doc/productivity.md

```diff
@@ -0,0 +1,165 @@
+Productivity Notes
+==================
+ 
+Table of Contents
+-----------------
+
+* [General](#general)
+   * [Cache compilations with `ccache`](#cache-compilations-with-ccache)
+   * [Disable features with `./configure`](#disable-features-with-configure)
+   * [Make use of your threads with `make -j`](#make-use-of-your-threads-with-make--j)
+   * [Multiple working directories with `git worktrees`](#multiple-working-directories-with-git-worktrees)
+* [Writing code](#writing-code)
+   * [Format C/C++/Protobuf diffs with `clang-format-diff.py`](#format-ccprotobuf-diffs-with-clang-format-diffpy)
+   * [Format Python diffs with `yapf-diff.py`](#format-python-diffs-with-yapf-diffpy)
+* [Rebasing/Merging code](#rebasingmerging-code)
+   * [More conflict context with `merge.conflictstyle diff3`](#more-conflict-context-with-mergeconflictstyle-diff3)
+* [Reviewing code](#reviewing-code)
+   * [Reduce mental load with `git diff` options](#reduce-mental-load-with-git-diff-options)
+   * [Reference PRs easily with `refspec`s](#reference-prs-easily-with-refspecs)
+   * [Diff the diffs with `git range-diff`](#diff-the-diffs-with-git-range-diff)
+
+General
+------
+
+### Cache compilations with `ccache`
+
+The easiest way to faster compile times is to cache compiles. `ccache` is a way to do so, from its description at the time of writing:
+
+> ccache is a compiler cache. It speeds up recompilation by caching the result of previous compilations and detecting when the same compilation is being done again. Supported languages are C, C++, Objective-C and Objective-C++.
+
+Install `ccache` through your distribution's package manager, and follow the symlinks method [here](https://ccache.samba.org/manual/latest.html#_run_modes) to set it up.
```

Comment:
I don't think it is required to do any fancy symlinking. Our configure will pick up ccache on its own.